packages/firebase_dynamic_links/lib/src/firebase_dynamic_links.dart

```diff
@@ -83,10 +86,12 @@ class FirebaseDynamicLinks {
   Future<dynamic> _handleMethod(MethodCall call) async {
     switch (call.method) {
       case "onLinkSuccess":
-        final Map<dynamic, dynamic> data =
-            call.arguments.cast<dynamic, dynamic>();
-        final PendingDynamicLinkData linkData =
-            getPendingDynamicLinkDataFromMap(data);
+        PendingDynamicLinkData linkData;
+        if (call.arguments != null) {
```

Comment:
info: `call.arguments.cast` crashes if the arguments are null. So added null check.